tx-builder/src/ckb/stake_smt.rs

```diff
@@ -1,130 +1,233 @@
 };
 
-use common::traits::tx_builder::IStakeSmtTxBuilder;
-use common::types::tx_builder::*;
-use common::utils::convert::*;
+use common::types::{
+    ckb_rpc_client::SearchKeyFilter,
+    tx_builder::{StakeSmtTypeIds, Staker as TxStaker},
+};
+use common::{
+    traits::smt::StakeSmtStorage,
+    types::{
+        ckb_rpc_client::Cell,
+        tx_builder::{Amount, CkbNetwork, Epoch, InStakeSmt, NonTopStakers, PrivateKey},
+    },
+};
+use common::{
+    traits::{ckb_rpc_client::CkbRpc, tx_builder::IStakeSmtTxBuilder},
+    types::ckb_rpc_client::{ScriptType, SearchKey},
+};
+use common::{
+    types::smt::{Staker as SmtStaker, UserAmount},
+    utils::convert::new_u128,
+};
+use molecule::prelude::Builder;
 
-use crate::ckb::define::constants::{INAUGURATION, TOKEN_BYTES};
-use crate::ckb::define::error::*;
-use crate::ckb::utils::cell_data::*;
+use crate::ckb::define::{
+    constants::{INAUGURATION, TOKEN_BYTES},
+    error::CkbTxErr,
+};
 
-pub struct StakeSmtTxBuilder {
-    _kicker:       PrivateKey,
-    current_epoch: Epoch,
-    quorum:        u16,
-    _stake_cells:  Vec<Cell>,
+use crate::ckb::utils::{
+    cell_collector::{collect_cells, get_stake_cell, get_withdraw_cell},
+    cell_data::{
+        stake_cell_data, stake_item, stake_smt_cell_data, token_cell_data, update_withdraw_data,
+    },
+    cell_dep::{
+        checkpoint_cell_dep, metadata_cell_dep, omni_lock_dep, secp256k1_lock_dep,
+        withdraw_lock_dep, xudt_type_dep,
+    },
+    omni::{omni_eth_address, omni_eth_witness_placeholder},
+    script::{always_success_lock, omni_eth_lock, stake_lock, stake_smt_type, xudt_type},
+    tx::balance_tx,
+};
+
+use super::utils::cell_dep::stake_dep;
+
+pub struct StakeSmtTxBuilder<C: CkbRpc, S: StakeSmtStorage + Send + Sync> {
+    ckb:               CkbNetwork<C>,
+    kicker:            PrivateKey,
+    current_epoch:     Epoch,
+    quorum:            u16,
+    stake_cells:       Vec<Cell>,
+    stake_smt_storage: S,
+    type_ids:          StakeSmtTypeIds,
 }
 
 #[async_trait]
-impl IStakeSmtTxBuilder for StakeSmtTxBuilder {
+impl<C: CkbRpc, S: StakeSmtStorage + Send + Sync> IStakeSmtTxBuilder<C, S>
+    for StakeSmtTxBuilder<C, S>
+{
     fn new(
-        _kicker: PrivateKey,
+        ckb: CkbNetwork<C>,
+        kicker: PrivateKey,
         current_epoch: Epoch,
+        type_ids: StakeSmtTypeIds,
         quorum: u16,
-        _stake_cells: Vec<Cell>,
+        stake_cells: Vec<Cell>,
+        stake_smt_storage: S,
     ) -> Self {
         Self {
-            _kicker,
+            ckb,
+            kicker,
             current_epoch,
             quorum,
-            _stake_cells,
+            stake_cells,
+            stake_smt_storage,
+            type_ids,
         }
     }
 
     async fn build_tx(&self) -> Result<(TransactionView, NonTopStakers)> {
-        // todo: get stake smt cell
-        let inputs = vec![];
+        let stake_lock = always_success_lock(&self.ckb.network_type); // todo: stake smt lock
```

Comment:
stake_lock -> stake_smt_lock. Because `stack_ smt_lock` is `always_sucess_lock`, useless comments should be removed.